README.md

```diff
@@ -1,13 +1,340 @@
 mafia
 =====
 
-> Provides protection against cabal swindling, robbing, injuring or sabotaging people with chopsticks.
+> Provides protection against cabal swindling, robbing, injuring or
+> sabotaging people with chopsticks.
 
 ![mafia](img/mafia.jpg)
 
+
 Overview
 --------
 
-`mafia` is an opinionated wrapper for `cabal` that makes working on Ambiata Haskell projects fun and easy.
+Mafia is an opinionated wrapper for Cabal that makes working on Haskell
+projects fun and easy.
 
 ![godfather](img/godfather.jpg)
+
+The central idea is that upon cloning a project, you should only have to
+run `mafia build` to get up and running for development. This will pull
+down any git submodules, create a cabal sandbox, install dependencies
+and then build everything, tests and benchmarks included.
+
+Cabal packages in git submodules are discovered automatically and made
+available as source dependencies. This is particularly useful if you
+have many internal dependencies that are not published to Hackage. It
+allows for development of a package and its dependencies all at once,
+without having to publish intermediate builds of libraries and without
+having to resort to a monorepo.
+
+All dependencies are cached globally in `$HOME/.ambiata/mafia/packages`
+using a nix-like hashing system so that for a given set of transitive
+dependencies, a package is only ever built once. Source dependencies are
+also cached, using their dependencies and source code as the hash. This
+is critical for Haskell development at scale. At Ambiata we have around
+150 Haskell packages which are being developed at a rapid pace, so it's
+crucial that we don't need to think about curating blessed sets of
+packages that we can cache as a single snapshot.
```

Comment:
Tempting to a fully fleshed out "motivation" section or separate doc, but then again maybe it's better not...